Update pull-release-test job to use k/release bazel image

saschagrunert opened the pull request:
This updates the bazel base image to the latest version.

Might be related to https://github.com/kubernetes/release/pull/974

saschagrunert commented:
I'm not sure if this is the best solution we can get. I also thought about bringing the gcc dependency somehow into the k/release bazel targets but this seems to have drawbacks as well. :thinking:

justaugustus commented:
/hold

justaugustus commented:
/assign

hoegaarden commented on config/jobs/kubernetes/release/release-config.yaml:
```diff
@@ -49,12 +49,14 @@ presubmits:
     spec:
       containers:
-      - image: launcher.gcr.io/google/bazel:0.29.1
+      - image: launcher.gcr.io/google/bazel:1.2.1
```
For what it's worth: Currently `2.0.0` seems to work as expected *on my machine*:tm: 🤷‍♂

hoegaarden replied:
I would also understand what those images are Where can I see there changelog or how they are built (docker(file), bazel, ...)? I could not answer that after a search. Can anyone help me out here?

saschagrunert replied:
It's this image: https://console.cloud.google.com/marketplace/details/google/bazel?q=bazel&id=e629cad7-9b87-4006-88ed-7725e80b1170

hoegaarden replied:
> It's this image: https://console.cloud.google.com/marketplace/details/google/bazel?q=bazel&id=e629cad7-9b87-4006-88ed-7725e80b1170

Really? The "Show pull command" button there shows `marketplace.gcr.io/google/bazel:latest`. And that's what's confusing me: what is the difference between all those bazel images. If I could see a changelog, how they are built, ... maybe it made more sense to me.

hoegaarden replied:
Oh yeah ... that's the image! Thanks!

saschagrunert replied:
I think the image will be built here:
https://github.com/GoogleCloudPlatform/container-definitions/tree/master/ubuntu1604_bazel

Wheres the image layer definitions are available here:
https://github.com/GoogleCloudPlatform/layer-definitions/tree/master/layers/ubuntu1604/bazel

justaugustus commented:
@ixdy @fejta @BenTheElder may have thoughts here.

BenTheElder commented on config/jobs/kubernetes/release/release-config.yaml:
```diff
@@ -49,12 +49,14 @@ presubmits:
+          - -c
+          - >
+            apt-get update &&
+            apt-get install -y build-essential &&
```
what is this about exactly? this smells like a flake / reproducibility nightmare.

saschagrunert replied:
The first issue was that golangci-lint does not error if it was not able to execute a dedicated linter. which has been fixed in https://github.com/golangci/golangci-lint/pull/890

Its reproducible by:
```bash
> podman run -v $(pwd):/w -w /w -it --entrypoint bash launcher.gcr.io/google/bazel:2.0.0
root@6509f76f5300:/w# bazel test @io_k8s_repo_infra//hack:verify-golangci-lint
...
INFO: Build completed successfully, 217 total actions
root@6509f76f5300:/w# cat bazel-testlogs/external/io_k8s_repo_infra/hack/verify-golangci-lint/test.log
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from @io_k8s_repo_infra//hack:verify-golangci-lint
-----------------------------------------------------------------------------
Verifying golangci-lint...
level=warning msg="[runner] Can't run linter goanalysis_metalinter: S1005: failed prerequisites: inspect@k8s.io/release/cmd/blocking-testgrid-tests, isgenerated@k8s.io/release/cmd/blocking-testgrid-tests"
level=warning msg="[runner] Can't run linter unused: buildssa: analysis skipped: errors in package: [/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/sandbox/processwrapper-sandbox/212/execroot/io_k8s_release/bazel-out/k8-fastbuild/bin/external/io_k8s_repo_infra/hack/verify-golangci-lint.runfiles/io_k8s_release/cmd/blocking-testgrid-tests/main.go:24:2: could not import net/http (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/net/http/client.go:13:2: could not import crypto/tls (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/crypto/tls/cipher_suites.go:16:2: could not import crypto/x509 (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/crypto/x509/verify.go:11:2: could not import net (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/net/cgo_linux.go:12:8: could not import C (cgo preprocessing failed))))) /root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/sandbox/processwrapper-sandbox/212/execroot/io_k8s_release/bazel-out/k8-fastbuild/bin/external/io_k8s_repo_infra/hack/verify-golangci-lint.runfiles/io_k8s_release/cmd/blocking-testgrid-tests/main.go:29:2: could not import k8s.io/test-infra/testgrid/config (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/sandbox/processwrapper-sandbox/212/execroot/io_k8s_release/_tmp/ca3cea22f615a4ff022845a3169046b1/home/go/pkg/mod/k8s.io/test-infra@v0.0.0-20190829230513-7ef687d80d22/testgrid/config/config.go:27:2: could not import cloud.google.com/go/storage (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/sandbox/processwrapper-sandbox/212/execroot/io_k8s_release/_tmp/ca3cea22f615a4ff022845a3169046b1/home/go/pkg/mod/cloud.google.com/go@v0.38.0/storage/acl.go:19:2: could not import net/http (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/net/http/client.go:13:2: could not import crypto/tls (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/crypto/tls/cipher_suites.go:16:2: could not import crypto/x509 (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/crypto/x509/verify.go:11:2: could not import net (/root/.cache/bazel/_bazel_root/1cce3e60ac2d51100ef178d6509a4972/external/go_sdk/src/net/cgo_linux.go:12:8: could not import C (cgo preprocessing failed)))))))]"
```

So yes, the linter fails to compile the package because of the missing c compiler. I would also prefer to add a dedicated build image from our side but I'm not sure where to host it.

justaugustus commented:
@saschagrunert -- gcr.io/k8s-staging-release-test/releng-ci-bazel:v20200110-v0.2.0-131-g137f2f6 is live. Please use the `latest` tag for now.

saschagrunert commented:
> @saschagrunert -- gcr.io/k8s-staging-release-test/releng-ci-bazel:v20200110-v0.2.0-131-g137f2f6 is live. Please use the `latest` tag for now.

Sure, I updated the image and rebased on top of the latest master branch.

saschagrunert commented on config/jobs/kubernetes/release/release-config.yaml:
```diff
@@ -41,17 +41,16 @@ presubmits:
+      - image: gcr.io/k8s-staging-release-test/releng-ci-bazel:latest
         command:
-        - ../test-infra/hack/bazel.sh
+        - bazel
```
I'm not sure if we need the `command` because the `entrypoint` of the image is already `bazel`. Making it explicit should not hurt here, right?

justaugustus replied:
Yes! Explicit please.
I _think_ Prow does a thing where it requires an explicit command.

BenTheElder replied:
pod-utils does not respect image entrypoints, you have to set `command` if `decorate` is set to `true`

saschagrunert replied:
Thanks for the hint, this is good to know. :)

justaugustus commented:
/lgtm
/approve
/hold cancel

saschagrunert commented on config/jobs/kubernetes/release/release-config.yaml:
```diff
@@ -41,17 +41,16 @@ presubmits:
   - name: pull-release-test
     always_run: true
     decorate: true
-    path_alias: k8s.io/release
```
@justaugustus the fact that I removed this is not problematic, right?
